content/en/docs/contribute/intermediate.md

```diff
@@ -379,25 +384,25 @@ After you decide which branch to start your work (or _base it on_, in Git
 terminology), use the following workflow to be sure your work is based on the
 most up-to-date version of that branch.
 
-1.  Fetch both the `upstream` and `origin` remotes. This updates your local
-    notion of what those branches contain, but does not change your local
-    branches at all.
+1.  Update your website fork (`origin`) from kubernetes/website (`upstream`).
+    Fetch both the `upstream` and `origin` remotes. Merge your `origin` with `upstream`
+    and push the changes to your `origin`. Your fork is now up-to-date with kubernetes/website.
 
       ```bash
-      git fetch upstream
+      git checkout master
       git fetch origin
+      git fetch upstream
+      git merge upstream/master
+      git push -f origin      
       ```
 
-2.  Create a new tracking branch based on the branch you decided is the most
+2.  Create a local working branch based on the branch you decided is the most
     appropriate. This example assumes you are using `master`.
 
       ```bash
-      git checkout -b <my_new_branch> upstream/master
```

Comment:
I like the original approach, of branching from `upstream/master`. I think it will help people who are new to Git avoid making a few kinds of mistake.

@aimeeu does that make sense? If you're keen on plain `git checkout -b <my_new_branch>`, can you explain why you prefer that?

Reply:
Honestly, I don't understand why the original text uses `git checkout -b <my_new_branch> upstream/master` . I've never seen that before and have never used it. We are creating a local branch that we push to our fork, so I just don't get the whole upstream/master part.  `upstream/master` is not used in the [Kubernetes GitHub Workflow](https://github.com/kubernetes/community/blob/master/contributors/guide/github-workflow.md) guide either... and thus more confusion.

Reply:
If the reader has mangled their local `master` branch, their PR will start off from that mangled local clone. Branching from `upstream/master` is safer because mangling `upstream/*` is much less likely.

I have _definitely_ seen people mangle their local `master` branch. Heck, I've done it (and noticed).

Reply:
@sftim - what kinds of mistakes? Maybe I'm making them and just don't know it. :)  I appreciate all feedback so I can make this as clear as possible.

Reply:
Thanks @sftim!

Reply:
Hope I've explained, can go into more detail if that's useful.

Reply:
To get more clarity on branching from upstream/master (beyond not wanting to branch from my own fork because I may have foobarred it, which is always a possibility...), I posted some [specific questions ](https://kubernetes.slack.com/archives/C1TU9EB9S/p1570736679005700) in #sig-contribex about the differences between the Docs and Community github workflows. Christoph Blecker provided very detailed answers. TL;DR because this doc's workflow does not
```
git checkout master
git fetch origin
git fetch upstream
git rebase upstream/master
git push -f origin
```
, you must **always** create a local working branch from upstream/master.  

To be honest, I haven't been using the workflow defined in this document. Rather, I've been using one very similar to the Community GitHub workflow: before each new PR, update my fork from upstream, then branch from my fork (passed to me by Zach and Jennifer Rondeau).  
```
git checkout master
git fetch origin
git fetch upstream
git merge upstream/master
git push -f origin
```

Reply:
I'm still doing a bit of a sucking-on-a-lemon face when I see `git push -f origin`. But that might well just be me!

Reply:
That's an image that's going to stick with me for the weekend!

Reply:
Changed back to `git checkout -b <my_new_branch> upstream/master` with explanation; added `--force-with-lease` to sections that previously used `-f`.